### PR TITLE
Ampers Sara S Ride Share Week4

### PR DESCRIPTION
**Pull request by CheerOnMars**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | For request trip, I decided to make the id of the new trip one higher than the max of the other trips. I thought that would be a better method than using the last id to ensure that a number wouldn’t be duplicated in case the ids weren’t in order. 
Describe any examples of composition that you encountered in this project, if any | An example of of composition in this project is that each of the rides has a driver and has a passenger. 
Describe a _nominal test_ that you wrote for this assignment. | When I tested the request_trip method, I created typically conditions to make sure we test that your code works normally. I had the first passenger be the one to request the trip. 
Describe an _edge case test_ that you wrote for this assignment. | I tested to confirm that it return an argument error if a trip was requested with an id that isn’t associated with a passenger.
Describe a concept that you gained more clarity on as you worked on this assignment. | One concept that I understood more was how to navigate through various classes and CSV files to pull the data fields that I was looking for.

### CODE REVIEW

**Comment by tildeee** [comment]
# Ride Share
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Used Git Regularly | :sparkles:
Answer comprehension questions | x
**Wave 1** |
`Trip` has a helper method to calculate duration | x
`Passenger` has a method to calculate total cost of all trips | x
`Driver` has a method to calculate total revenue | x
Complete tests | x
**Wave 2** |
`TripDispatcher` has a new method to create trips | x
Creating a trip in `TripDispatcher` relies on methods in `Driver` and `Passenger` to modify their own attributes | x
Methods from wave 1 work on `Driver`s and `Passenger`s with incomplete trips | untested
Complete tests | x
**Wave 3** (optional) |
Complex logic was correctly implemented and tested | 
Appropriate use of Ruby's `Date` | 
Appropriate helper methods were made to help with complex logic | 
Complete tests | 
**Overall** | 

Good job overall!

You tended to use `.each` iteration to calculate totals/sum things, even though `.sum` is available. Honestly, I don't believe that you ALWAYS need to use fancy Enumerable methods like `.sum` over `.each` iteration every time, but I want to make sure you know that those are possibilities.

Your test cases are so thorough and clean, it makes me so incredibly happy :)

You didn't test for one case consistently though: instances of `Trip` can have an `end_time` of `nil` and while your code handles it, you don't have a test case for that scenario.

Nicely done!

File: lib/passenger.rb [hunk 1]
```diff
@@ -20,5 +22,30 @@ def get_drivers
...
+      return total_spent.to_f
+    end
+
+    def calc_total_duration
+      total_duration = 0
+      @trips.each do |trip|
+        if trip.end_time != nil
+          total_duration += trip.duration
+        else
+          total_duration
```
Comment (tildeee) [inline lib/passenger.rb:44]: just a weird small note: if `trip.end_time != nil` is false and the code execution gets into this `else`, just calling `total_duration` won't actually do anything :) which is harmless, but not helpful.
